aws-amplify/amplify-category-api
# merge main into fork nov 9

#### Description of changes
Merge `main` into `feature/al-main-fork` and fixup generated files, etc.

Merge the following commits:
- fix(api): make id optional if not a string type
- feat(api): add arrays and objects support for rds datasource
- fix api extract and lint issues
- chore(api): update rds layer to version 8
- feat(api): rds has many support
- add files
- add e2e tests
- add unit tests
- fix lint and api extract
- add validation for references fields match
- feat: add debug mode env variable
- hasone and belongs to support
- add happy case unit tests
- add e2e tests
- lint and api extract fix
- update layer version
- Preserve RDS schema edits for JSON fields, test error logging (#1783)
- Check for directives not supported in RDS yet (#1795)
- address build errors
- feat(api): custom queries support using sql directive
- fix model directive tests
- fix relational directive unit tests
- fix transformer unit tests
- refactor reading sql statement
- fix api tests
- feat(api): specify include or exclude tables option to generate schema
- lint fixes
- lint fix
- address code review comments
- fix: Fix schema import for RDS clusters in VPCs
- update snapshot
- chore(api): merge main into rds feature branch (#1837)
- Revert "chore(api): merge main into rds feature branch (#1837)"
- feat(api): add vpc endpoints for ssm
- add vpc endpoints as env variable to lambda
- test fixes
- add vpc config unit test
- feat: add refersTo directive transformer for model renaming (#1830)
- feat: infer refersTo for models in the import flow and preserve edits during schema regeneration (#1845)
- fix(graphql): disable default value validation for rds
- chore: lint fixes
- chore: use fixed jest dep versions
- chore: ts-jest to have same version as jest
- lint and api extract fix
- version fixes
- test(graphql): restrict public access to the database
- test(graphql): modify all e2e tests
- lint fix
- fix e2e core send yes
- setup database in app region
- fix vpc step for generate schema
- increase nooutput timeout for vpc flow
- fix test errors
- add delay for vpc endpoint to go live
- update layer version
- refactor to helper methods
- feat: refersTo supports field name mappings on RDS models (#1865)
- feat(api): refactor auth transformer to use vtl generator factory pattern
- fix lint and api extract
- fix unit test
- fix unsupported directives error
- correct typo
- chore: update RDS layer version
- feat: refersTo transformer to support field name mappings on RDS models
- test: add e2e tests for field mappings
- feat: infer and preserve field mappings in schema
- fix: check for duplicate model name mappings
- chore: update API files
- test: update e2e tests
- fix: handle field names starting with number
- fix: change field mapping slot to preAuth
- fix: handle non-alphabetic names
- chore: update test description
- feat(api): rds auth model level rules
- add unit tests
- fix sql and relational support
- refactor mutation helpers
- fix snapshots
- add e2e test wrapper
- lint and api extract fixes
- update snapshots
- minor syntax fixes
- fix(api): iam rule correct cognito pool id
- feat: Support RDS Proxy schema imports (#1917)
- add test and minor fix
- lint and snapshot fix
- remove console logs
- reset auth test coverage
- handle null on identity pool id
- fix prettier linting
- test: Fix RDS e2e test
- chore: e2e doc update
- chore: lint fix publish-notification-lambda/README.md
- fix: Update rds transformers to accommodate ConfiguredAuthProviders change
- lint fix
- chore: Reorder generateAuthRulesFromRoles arguments
- test(api): e2e test for iam auth on rds
- fix rebase and lint issues
- add missing import
- feat(api): postgres import workflow
- feat(api): prompt changes to add postgres support
- add e2e and unit tests
- add logs access to schema inspector lambda
- add more unit tests to pg datatype mapping
- add unit tests for all scalar types
- test: add userpool, OIDC and custom claims auth fixes and tests for RDS (MySQL) (#1920)
- fix rebase issues
- fix e2e build test
- import type errors
- lint and api extract fixes
- fix(api): do not allow auth directive on rds model fields
- lint fix
- fix: relational auth cases; preserve auth rule schema edits, added multi-auth e2e tests (#1965)
- chore: Consolidate subnet & AZ configs in VPC config (#1961)
- update tests
- fix(api): iam support for admin roles with auth utils
- feat: Add SQL database support to AmplifyGraphqlApi construct (#1986)
- add unit test
- fix lint
- feat(api): add enum support for postgres import
- feat: add postgres engine and update types as needed (#1979)
- lint and api extract fix
- remove path
- chore: merge main to rds
- chore: update rds lambda layer version
- test(api): postgres relational, sql and complex types
- lint fix
- chore: bump rds layer version
- test(api): pg e2e auth tests
- test: E2E tests for SQL support in CDK construct (#2012)
- chore: Rename DatasourceType to DataSourceType (#2018)
- fix: Make VPC configuration optional (#2021)
- test: add Postgres E2E tests (#2005)
- refactor e2e code
- remove command logging
- feat: add graphqlSchemaFromRDSSchema for unauth schema import (#2004)
- feat: add postgres support to graphqlSchemaFromRDSSchema (#2015)
- feat: validate imported schema (#2031)
- chore: disable failing auth e2e tests for RDS (#2035)
- correct DBType, DataSourceType imports
- Change signature of TransformerDataSourceManager.add to conform to TransformerDataSourceManagerProvider
- WIP: Mechanical type updates
- WIP fixing functional merge conflicts
- all compile errors fixed
- Temporarily reduce unit test coverage thresholds for amplify-graphql-model-transformer
- fix: only access .rows on knex schema query result (#2037)
- test: fix unit tests
- Temporarily disable deploy-velocity tests
- fix e2e tests
- test: add create-env script to base e2e tests
- pr feedback: explicitly handle PG data type
- Disable eslint rules in nexpect for easier IDE viewing
- Add db engine to rds-model-v2 'describe' label
- add wait after disabling inbound rules
- put wait in right place
- fix: PostgreSQL introspection logic (#2043)
- chore: add preview warning for modelDataSourceDefinition

##### CDK / CloudFormation Parameters Changed
Probably, but not as part of this PR, part of the underlying changes.

#### Issue #, if available
N/A

#### Description of how you validated changes
Running unit + e2e tests in CI

#### Checklist
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
